coblocks-loader.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * Plugin Name: CoBlocks
+ * Description: Custom WordPress.com CoBlocks fork.
+ *
+ * Our CoBlocks fork is a mu-plugin and it's loaded by default using this loader.
+ * This includes additional logic to check if another version of CoBlocks is
+ * present. If that's the case, it'll take precedence over this one.
+ **/
+if ( defined( 'IS_ATOMIC' ) && IS_ATOMIC ) {
```

Comment:
This constant may be defined by another mu-plugin and I don't think we have guarantees about load order.

I'd say we should invert this, define a function and always call it on plugins loaded. Inside the function we short-circuit to skip handling if the constant is set.

We should be able to use namespaces on all of the platforms we need to support as well, so this might be `A8c_CoBlocks::plugins_loaded()`.

Reply:
> This constant may be defined by another mu-plugin and I don't think we have guarantees about load order.

Reply:
> This constant may be defined by another mu-plugin and I don't think we have guarantees about load order.

Isn't this a global constant? It'd be defined once (as `TRUE`) somewhere in the app init code when running in the AT env (`wp-config.php`?) and then checked against here and anywhere else that depends on it, AFAIU? I'm not sure I understand your concern about the definition and load order, could you elaborate / exemplify?

Reply:
I didn't fully understand the reasoning but I can see why it'd be safer to have it inside the `load-coblocks` function, specially if we move this to another env, like the WET 👍🏻

Reply:
>  skip handling if the constant is set.

I guess you meant "not" set? We're trying to make sure we're in the AT env, not the other way around. If we want to short-circuit - and if I got the gist of your idea correctly - we'd need something like:

```
  function load_coblocks() {
    if ( ! defined( 'IS_ATOMIC' ) || ! IS_ATOMIC ) return;

    $load_mu_coblocks = function() { require_once( WPMU_PLUGIN_DIR . '/coblocks/class-coblocks.php' ); };

    $is_activating_coblocks = (isset($_GET['action']) && isset($_GET['plugin'])) &&
      ($_GET['action'] == 'activate' && $_GET['plugin'] == 'coblocks/class-coblocks.php');

    $official_coblocks_present = in_array('coblocks/class-coblocks.php', get_option('active_plugins'), TRUE);

    if (!$is_activating_coblocks && !$official_coblocks_present) {
      $load_mu_coblocks();
    }
  }
```
Right?

Reply:
Yep, that looks like what I had in mind 👍